### ISSUE TITLE
2D instead of 1D array from newArrayOfSize(1).get(0)

### ISSUE THREAD

**Issue by epoell** [eclipse/xtext-extras#394]
The expression
```
val x = newArrayOfSize(1).get(0)
```
compiles to an two dimensional array in Java.
```
final Object x = new Object[1][0];
```
One would expect the array to be one dimensional with only one element. Thus this might cause errors in Xtend as types might be incompatible.

If the array is initialised first and then get is called, the array is one dimensional.
Xtend:
```
val a = newArrayOfSize(1)
val x = a.get(0)
```
Java:
```
final Object[] a = new Object[1];
final Object x = a[0];
```

**Comment by cdietrich** [comment]
```
@Inline("new $3[$1]")
```

=> i wonder why `$3` gets `new Object[1]`

**Comment by cdietrich** [comment]
```
newArrayOfSize(1)
```
=> `new Object[1]`

```
somearray.get(0)
```
=> `somearray[0]`

=> imho works as designed

**Comment by cdietrich** [comment]
@szarnekow this could be a duplicate / reincarnation of 
https://github.com/eclipse/xtext-lib/issues/60

any ideas? do you think we can always set () in this case?

**Comment by cdietrich** [comment]
i wonder if we can do something like

```
boolean found = false;
		if (call.eContainer() instanceof XMemberFeatureCall) {
			JvmAnnotationReference inlineAnnotation2 = expressionHelper.findInlineAnnotation(((XMemberFeatureCall)call.eContainer()));
			if (inlineAnnotation2 != null) {
				found = true;
				b.append("(");
			}
		}
```

**Comment by cdietrich** [comment]
or should we be even more general like

```
	protected boolean isNestedInlineCall(XAbstractFeatureCall call) {
		if (call.eContainer() instanceof XMemberFeatureCall) {
			return true;
		}
		return false;
	}
```
which could fix https://github.com/eclipse/xtext-lib/issues/60 too

**Comment by cdietrich** [comment]
Fixed in 2.18
